server/tests/roadmaps.test.ts

```diff
@@ -2,113 +2,151 @@ import chai, { assert, expect } from 'chai';
+      const fetchedAgent = await User.query()
+        .where('username', agentData.username)
+        .withGraphFetched('roadmaps');
       expect(res.status).to.equal(200);
-      expect(res.body.length).to.be.greaterThan(0);
-      expect(res.body[0]).to.have.property('id');
-      expect(res.body[0]).to.have.property('name');
-      expect(res.body[0]).to.have.property('description');
-      expect(res.body[0]).to.have.property('tasks');
+      expect(res.body.length).to.equal(fetchedAgent[0].roadmaps.length);
+      expect(res.body[0]).to.have.keys('id', 'name', 'description', 'tasks');
     });
 
     it('Should return all roadmaps, eager loaded', async function () {
       const res = await loggedInAgent.get('/roadmaps?eager=1');
+      const fetchedAgent = await User.query()
+        .where('username', agentData.username)
+        .withGraphFetched('roadmaps');
       expect(res.status).to.equal(200);
-      expect(res.body.length).to.be.greaterThan(0);
-      expect(res.body[0]).to.have.property('id');
-      expect(res.body[0]).to.have.property('name');
-      expect(res.body[0]).to.have.property('description');
-      expect(res.body[0]).to.have.property('tasks');
-      expect(res.body[0].tasks[0]).to.have.property('id');
-      expect(res.body[0].tasks[0]).to.have.property('name');
-      expect(res.body[0].tasks[0]).to.have.property('description');
-      expect(res.body[0].tasks[0]).to.have.property('completed');
-      expect(res.body[0].tasks[0]).to.have.property('createdAt');
-      expect(res.body[0].tasks[0]).to.have.property('roadmapId');
-      expect(res.body[0].tasks[0]).to.have.property('createdByUser');
-      expect(res.body[0].tasks[0]).to.have.property('ratings');
+      expect(res.body.length).to.equal(fetchedAgent[0].roadmaps.length);
+      expect(res.body[0]).to.have.keys(
+        'id',
+        'name',
+        'description',
+        'tasks',
+        'integrations',
+      );
+      expect(res.body[0].tasks[0]).to.have.keys(
+        'id',
+        'name',
+        'description',
+        'completed',
+        'createdAt',
+        'importedFrom',
+        'externalId',
+        'externalLink',
+        'ratings',
+        'createdByUser',
+        'roadmapId',
+      );
     });
   });
 
   describe('POST /roadmaps/', function () {
     it('Should add new roadmap', async function () {
+      const newRoadmap = {
+        name: 'TestRoadmap',
+        description: 'TestDescription',
+      };
+
       const before = await loggedInAgent.get('/roadmaps/');
       const res = await loggedInAgent
         .post('/roadmaps/')
         .type('json')
-        .send({ name: 'TestRoadmap', description: 'TestDescription' });
+        .send(newRoadmap);
       const after = await loggedInAgent.get('/roadmaps/');
+
+      const fetchedAgent = await User.query()
+        .where('username', agentData.username)
+        .withGraphFetched('roadmaps');
+
+      const addedRoadmapDb = fetchedAgent[0].roadmaps.find(
```

Comment:
Ah, that's a smarter way to do it so you don't have to mess with arrays there. I'll fix it!

Reply:
I changed it to use findOne instead of the original where:
```
      const fetchedAgent = await User.query()
        .findOne('username', agentData.username)
        .withGraphFetched('roadmaps');
```
so there's one less line of code now!

Reply:
Even better! 🙌